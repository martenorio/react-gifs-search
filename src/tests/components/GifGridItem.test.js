import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme'

describe('Test gifgriditem', () => {
    const title = 'titulo';
    const url = 'https://www.udemy.com/staticx/udemy/images/v7/logo-udemy-inverted.svg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('should show <GifGridItem>', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('should have a paragraph', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    })
    test('should have image equal to url and alt of the props', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })
    test('should have animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
    
    
})
