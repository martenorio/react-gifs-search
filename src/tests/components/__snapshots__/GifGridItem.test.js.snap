// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test gifgriditem should show <GifGridItem> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://www.udemy.com/staticx/udemy/images/v7/logo-udemy-inverted.svg"
  />
  <p>
    titulo
  </p>
</div>
`;
